src/server/editorServices.ts

```diff
@@ -783,10 +784,13 @@ namespace ts.server {
         /*@internal*/
         private packageJsonFilesMap: ESMap<Path, FileWatcher> | undefined;
 
+        /*@internal*/
+        readonly session: Session<unknown> | undefined;

```

Comment:
I think you should wire TMessage thing here and PluginCreateInfo too,

Reply:
I thought the `unknown` is better:

If we add type parameter `TMessage ` into `editorService`, We should also add the type parameter into `Project` and etc. Too many changes.

And we are not really care about the type of message in `editorService`. We just forward session into pluin.

And the `handlers` or `addProtocolHandler` does not have `TMessage ` too. Thay are raw string.

Reply:
If that’s the case we shouldn’t add it to project service options either

Reply:
Agreed.